ext/liquid_c/block.c

```diff
@@ -222,11 +219,34 @@ static tag_markup_t internal_block_body_parse(block_body_t *body, parse_context_
                     break;
                 }
 
-                VALUE tag_name = rb_enc_str_new(name_start, name_end - name_start, utf8_encoding);
-                VALUE tag_class = rb_funcall(tag_registry, intern_square_brackets, 1, tag_name);
+                if ((name_len == 4 && strncmp(name_start, "else", 4) == 0) || (name_len == 5 && strncmp(name_start, "endif", 5) == 0)) {
+                    VALUE str = rb_enc_str_new(name_start, name_len, utf8_encoding);
+                    unknown_tag = (tag_markup_t) { str, str };
+                    goto loop_break;
+                }
 
                 const char *markup_start = read_while(name_end, end, rb_isspace);
                 VALUE markup = rb_enc_str_new(markup_start, end - markup_start, utf8_encoding);
+                VALUE tag_name = rb_enc_str_new(name_start, name_end - name_start, utf8_encoding);
+
+                if (name_len == 2 && strncmp(name_start, "if", 2) == 0) {
+                    unknown_tag = parse_if_tag(markup, body, parse_context);
+                    if (unknown_tag.name != Qnil) {
+                        goto loop_break;
+                    }
+                    render_score_increment += 1;
+                    body->as.intermediate.blank = false;
+                    break;
+                } else if (
```

Comment:
handling unknown tags here could get messy as we introduce other tags, need a better abstraction for this.

Reply:
I was thinking of something along the lines of https://github.com/Shopify/liquid-c/pull/96 so we could pick the right parsing handler from ruby land.